File: EnrollmentManager.API/Models/Student.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EnrollmentManager.API.Models;

public class Student
{
    [Key, ForeignKey("User")]
    public int UserId { get; set; }

    // O CPF passa a ser opcional (string?) caso seja um aluno internacional
    [MaxLength(14)] 
    public string? Cpf { get; set; }

    // Campos para alunos internacionais
    [MaxLength(50)] 
    public string? PassportNumber { get; set; }

    [MaxLength(50)] 
    public string? Nationality { get; set; }

    public DateTime BirthDate { get; set; }
    
    [MaxLength(20)] 
    public string Phone { get; set; } = string.Empty;
    
    [MaxLength(200)] 
    public string Address { get; set; } = string.Empty;
    
    [Required] 
    [MaxLength(50)] 
    public string RegistrationNumber { get; set; } = string.Empty;

    // Propriedade de navegação
    public User User { get; set; } = null!;
}

File: frontend/src/app/models/student.ts
export interface StudentCreateDto {
  userId: number;
  cpf?: string | null;
  passportNumber?: string | null;
  nationality: string;
  birthDate: string;
  phone: string;
  address: string;
  formatIds: number[];
}

export interface UserStudentCreateDto extends Omit<StudentCreateDto, 'userId'> {
  userName: string;
  email: string;
  password: string;
}

export interface StudentUpdateDto {
  cpf?: string | null;
  passportNumber?: string | null;
  nationality?: string | null;
  birthDate: string;
  phone: string;
  address: string;
  formatIds: number[];
}

export interface StudentResponseDto {
  userId: number;
  userName: string;
  email: string;
  cpf?: string | null;
  passportNumber?: string | null;
  nationality?: string | null;
  birthDate: string;
  phone: string;
  address: string;
  registrationNumber: string;
  formatIds: number[];
}
